impeller/entity/geometry.cc

```diff
@@ -936,25 +933,125 @@ GeometryResult PointFieldGeometry::GetPositionBufferCPU(
       vtx_builder.AppendVertex({pt2});
     }
   }
+  return vtx_builder;
+}
+
+GeometryResult PointFieldGeometry::GetPositionBufferGPU(
+    const ContentContext& renderer,
+    const Entity& entity,
+    RenderPass& pass,
+    std::optional<Rect> texture_coverage,
+    std::optional<Matrix> effect_transform) {
+  FML_DCHECK(renderer.GetDeviceCapabilities().SupportsCompute());
+  if (radius_ < 0.0) {
+    return {};
+  }
+  auto determinant = entity.GetTransformation().GetDeterminant();
+  if (determinant == 0) {
+    return {};
+  }
+
+  Scalar min_size = 1.0f / sqrt(std::abs(determinant));
+  Scalar radius = std::max(radius_, min_size);
+
+  auto vertices_per_geom = ComputeCircleDivisions(
+      entity.GetTransformation().GetMaxBasisLength() * radius, round_);
+
+  auto points_per_circle = 3 + (vertices_per_geom - 3) * 3;
+  auto total = points_per_circle * points_.size();
+
+  auto cmd_buffer = renderer.GetContext()->CreateCommandBuffer();
+  auto compute_pass = cmd_buffer->CreateComputePass();
+  auto& host_buffer = compute_pass->GetTransientsBuffer();
+
+  auto points_data =
+      host_buffer.Emplace(points_.data(), points_.size() * sizeof(Point),
+                          DefaultUniformAlignment());
+
+  DeviceBufferDescriptor buffer_desc;
+  buffer_desc.size = total * sizeof(Point);
+  buffer_desc.storage_mode = StorageMode::kDevicePrivate;
+
+  auto geometry_buffer = renderer.GetContext()
```

Comment:
Small discovery, if I use buffers allocated from the compute passes transients buffer, then the second compute pass doesn't correctly synchronize on the completion on the first. I suspect metal is tracking buffer inputs/outputs to do automatic fences and the singular buffer defeats that detection.